batch7crud-roll9/src/main/java/com/lftechnology/batch7crud/dao/StudentDAO.java

```diff
@@ -0,0 +1,127 @@
+package com.lftechnology.batch7crud.dao;
+
+import com.lftechnology.batch7crud.constants.Query;
+import com.lftechnology.batch7crud.entity.Student;
+import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.util.DBConnection;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by sanjay on 1/14/16.
+ */
+public class StudentDAO {
+  private static final Logger LOGGER = Logger.getLogger(StudentDAO.class.getName());
+
+  public void insert(Student s) throws DataException//get the object
+  {
+    try (Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.INSERT);) {
+      pstmt.setString(1, s.getFirstName());
+      pstmt.setString(2, s.getMiddleName());
+      pstmt.setString(3, s.getLastName());
+      pstmt.setString(4, s.getAddress());
+      pstmt.setInt(5, s.getGrade());
+      pstmt.executeUpdate();
+      conn.close();
+    } catch (SQLException ex) {
+      LOGGER.log(Level.SEVERE, ex.getMessage(), ex);
+      throw new DataException();
+    }
+  }
+
+  public List<Student> fetch(int page, int limit) throws DataException {
+    try(Connection conn = DBConnection.getConnection(); PreparedStatement pstmt =  conn.prepareStatement(Query.LIST)) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) "" is provided externally to the method and not sanitized before use. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2077)